### PR TITLE
feat(server): Add metrics

### LINKED ISSUE #637: feat: introduce a dragonfly_version metric
As the title says, please add a metric like `dragonfly_version`. Could even combine that, with exposing an additional `dragonfly_version_update` or something, if the integrated update check found an update.

Memcached for example exports the following
```
# HELP memcached_version The version of this memcached server.
# TYPE memcached_version gauge
memcached_version{version="1.6.17"} 1
```

Another example from Victoria Metrics, which would probably work better, as the output of `dragonfly --version` is relatively long, would be something like
```
vm_app_version{version="vmstorage-20221220-230939-tags-v1.85.3-cluster-0-g30b15111a", short_version="v1.85.3-cluster"} 1
```

So, for Dragonfly, I could see something like
```
dragonfly_version{version="v0.13.1-f7e70bc45ad045e6ea49f6b1118df554189aaf2d", short_version="v0.13.1"} 1
```
---
In the same go, I'd even suggest to deprecate the `dragonfly_up` metric, as that's redundant to querying `up{job="argocd-dragonfly-metrics"}` or `up{pod="argocd-dragonfly-0"}` (direct examples from my setup) directly.

### PR DESCRIPTION
**Pull request by ashotland**
Signed-off-by: ashotland <ashotland@gmail.com>

Fixes #637  

Added:
- version
- hits/misses
- rss memory 

Screenshot from 2023-01-03 17-04-47

### CODE REVIEW

**Comment by romange** [comment]
Please note the naming conventions for prometheus:
https://prometheus.io/docs/practices/naming/

specifically add _total to hits and misses imho.
Also, pls change the commit description to has "Fixes #..." to link it to
the issue.

File: src/server/server_family.cc [hunk 1]
```diff
@@ -665,9 +668,19 @@ void PrintPrometheusMetrics(const Metrics& m, StringResponse* resp) {
...
+                              &resp->body());
+  } else {
+    LOG_FIRST_N(ERROR, 10) << "Error fetching /proc/self/status stats. error "
+                             << sdata_res.error().message();
+  }                           
 
+  // Stats metrics
   AppendMetricWithoutLabels("commands_processed_total", "", m.conn_stats.command_cnt,
                             MetricType::COUNTER, &resp->body());
+  AppendMetricWithoutLabels("keyspace_hits", "", m.events.hits, MetricType::COUNTER, &resp->body());
```
Comment (romange) [inline src/server/server_family.cc:682]: this should end with `_total` at the end.

File: src/server/server_family.cc [hunk 2]
```diff
@@ -646,8 +646,10 @@ void AppendMetricWithoutLabels(string_view name, string_view help, const absl::A
 void PrintPrometheusMetrics(const Metrics& m, StringResponse* resp) {
   // Server metrics
   AppendMetricWithoutLabels("up", "", 1, MetricType::GAUGE, &resp->body());
```
Comment (ashotland) [inline src/server/server_family.cc:648]: Remove

**Comment by romange** [comment]
You should setup your presubmit checks - you have trailing whitespace in the commit

**Review by romange — APPROVED** [review]

**Pull request merged**